### PR TITLE
Store bumps as part of data accounts instead of passing in ix

### PR DESCRIPTION
**Pull request by botdad**
I think this cleans up the calls by quite a bit so lets consider it for the deploy. Comments welcome

### CODE REVIEW

**Review by rweald — COMMENTED** [review]
I like it! Just having to pass around less things into the swap JS calls is super nice, plus less likely to get upstream mistakes.

File: tests/shared.ts [hunk 1]
```diff
@@ -122,15 +121,14 @@ export const fixture = async (
...
-      canSwap.programId
-    );
+  const [wrappedTokenAccountAuthority] = await PublicKey.findProgramAddress(
+    [
+      WRAPPED_TOKEN_OWNER_AUTHORITY_PDA_SEED,
+      canonicalMint.toBuffer(),
+      wrappedMint.toBuffer(),
+    ],
+    canSwap.programId
+  );
```
Comment (jonorally) [inline tests/shared.ts:132]: We don't need to adjust the `8 + 68` below?

**Review by jonorally — APPROVED** [review]
+1, much cleaner!

File: tests/shared.ts (hunk 1 above)
Comment (botdad) [reply tests/shared.ts:132]: I accidentally included this change in the previous PR, this byte count is correct

**Pull request merged**